### PR TITLE
Escape control characters in metadata

### PR DESCRIPTION
**Pull request by mkosiarc**
CLOUDBLD-11243

Signed-off-by: mkosiarc <mkosiarc@redhat.com>

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- [x] Python type annotations added to new code
- n/a JSON/YAML configuration changes are updated in the relevant schema
- [x] Changes to metadata also update the documentation for the metadata
- n/a Pull request has a link to an osbs-docs PR for user documentation updates
- n/a New feature can be disabled from a configuration file

### CODE REVIEW

**Comment by ben-alkov** [comment]
@mkosiarc; Should the commit message mention *why* we're escaping control characters in metadata (and which metadata)?

**Comment by mkosiarc** [comment]
> @mkosiarc; Should the commit message mention _why_ we're escaping control characters in metadata (and which metadata)?

Updated the commit slightly, more information can be found in the linked ticket.

**Review by rcerven — APPROVED** [review]

File: atomic_reactor/plugins/koji_import.py [hunk 1]
```diff
@@ -801,3 +803,19 @@ def _update_build(self, build):
             'release': nvr['release'],
             'source': self.workflow.data.koji_source_source_url,
         })
+
+
+def escape_non_printable_chars(koji_metadata: Dict[str, Any]) -> Dict[str, Any]:
```
Comment (MartinBasti) [inline atomic_reactor/plugins/koji_import.py:808]: Don't we usually define functions at the top before classes?

**Comment by ben-alkov** [comment]
> Updated the commit slightly, more information can be found in the linked ticket.

Yes, but not everyone in the world has access to the linked ticket

File: atomic_reactor/plugins/koji_import.py (hunk 1 above)
Comment (mkosiarc) [reply atomic_reactor/plugins/koji_import.py]: moved

**Comment by mkosiarc** [comment]
Should I include a small note in koji.md docs?

**Comment by rcerven** [comment]
> Should I include a small note in koji.md docs?

not really, it is brew internal thing

**Review by MartinBasti — APPROVED** [review]

**Pull request merged**